File: app/src/main/java/com/amandariu/tagger/demo/utils/AlertUtils.java
package com.amandariu.tagger.demo.utils;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.amandariu.tagger.R;

/**
 * Various helper methods for displaying alert messages to the end user.
 *
 * @author amandariu (11/5/17)
 */
public abstract class AlertUtils {

    /**
     * Displays an alert with a title of "Error" and whatever message provided.
     *
     * @param msg The message to display
     */
    public static void displayErrorMessage(@NonNull Context context, @NonNull String msg) {
        String title = context.getResources().getString(R.string.error);
        displayAlertMessage(context, title, msg);
    }

    /**
     * Pop a simple alert dialog to the end-user
     *
     * @param title The title of the alert dialog
     * @param msg   The message to display
     */
    public static void displayAlertMessage(@NonNull Context context,
                                           @Nullable String title, @NonNull String msg) {

        AlertDialog.Builder builder = new AlertDialog.Builder(context)
                .setMessage(msg)
                .setNeutralButton(android.R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                });

        if (title != null) {
            builder.setTitle(title);
        }

        builder.create().show();
    }
}

File: app/src/main/java/com/amandariu/tagger/demo/utils/AlertUtils.kt
package com.amandariu.tagger.demo.utils

import android.app.AlertDialog
import android.content.Context
import com.amandariu.tagger.demo.R

/**
 * Displays an alert with a title of "Error" and whatever message provided.
 *
 * @param [context] The active context.
 * @param [msg] The message to display
 */
fun displayErrorMessage(context: Context, msg: String) {
    val title = context.resources.getString(R.string.error)
    displayAlertMessage(context, title, msg)
}

/**
 * Pop a simple alert dialog to the end-user.
 *
 * @param [context] The active context.
 * @param [title] The title of the alert dialog.
 * @param [msg] The message to display.
 */
fun displayAlertMessage(context: Context, title: String?, msg: String) {
    val builder = AlertDialog.Builder(context)
            .setMessage(msg)
            .setNeutralButton(android.R.string.ok) { dialog, _ -> dialog.dismiss() }
    title?.let {
        builder.setTitle(it)
    }
    builder.create().show()
}